File: 0001-two-sum/0001-two-sum.java
// class Solution
// {
//     public int[] twoSum(int[] nums, int target)
//     {
//        for(int i=0;i<nums.length;i++){
//            for(int j=i+1;j<nums.length;j++){
//                if(nums[j]==target-nums[i]){
//                     return new int[]{i,j};
//                }
//            }
//        }
//          return null;
//     }
// }

class Solution
{
    public int[] twoSum(int[] nums, int target)
    {
        int[] result=new int[2];
        Map<Integer,Integer> map=new HashMap<>();
       for(int i=0;i<nums.length;i++){
        if(!map.containsKey(target-nums[i])){
            map.put(nums[i],i);
        }else{
            result[1]=i;
            result[0]=map.get(target-nums[i]);
            return result;
        }
       }
         return result;
    }
}

File: 0001-two-sum/0001-two-sum.js
;

var twoSum = function(nums, target) {
    let ans = [-1, -1];
    let map = new Map();
    for (let i = 0; i < nums.length; i++) {
        let value = nums[i];
        let check = target - value;
        if (map.has(check)) {
            ans[0] = map.get(check);
            ans[1] = i;
        } else {
            map.set(nums[i], i);
        }
    }
    return ans;
};
